modules/rust_examples/RsExampleUser/src/eventually_generated.rs

```diff
@@ -23,51 +19,74 @@ impl ExampleClient {
             "key": key,
         });
         let blob = self
-            .raw_publisher
+            .runtime
             .call_command("their_example", "uses_something", &args);
         let return_value: bool = ::serde_json::from_value(blob)
             .map_err(|_| everestrs::Error::InvalidArgument("return_value"))?;
         Ok(return_value)
     }
 }
 
-pub trait Module: Sized {
-    fn on_ready(&self) {}
-    fn main(&self) -> &dyn ExampleUserService;
-    fn their_example_subscriber(&self) -> &dyn ExampleSubscriber;
+pub trait ExampleSubscriber: Sync + Send {
+    fn on_max_current(&self, pub_impl: &ExamplePublisher, value: f64);
 }
 
-/// We want the user to just implement the `Module` trait above to get access to everything that
-/// EVerest has to offer, however for the `everestrs` library, we have to implement the
-/// `GenericModule`. This thin wrapper does the translation between the generic module and the
-/// specific implementation provided by the user.
-pub struct GenericToSpecificModuleProxy<T: Module>(T);
+pub trait OnReadySubscriber: Sync + Send {
+    fn on_ready(&self);
+}
+
+/// The struct holding the subscriber implementation.
+struct ModuleSubscriber {
+    /// The server subscribers.
+    their_example: Arc<dyn ExampleSubscriber>,
```

Comment:
The subscriber are arcs - if the user want to maintain an own handle - fine - if not we will take them